feat(i18n): add Tunisian Arabic (ar-TN) translation

nourzakhama2003 opened the pull request:
> [!IMPORTANT]
>
> 1. Make sure you have read our [contribution guidelines](https://github.com/langgenius/dify/blob/main/CONTRIBUTING.md)
> 1. Ensure there is an associated issue and you have been assigned to it
> 1. Use the correct syntax to link this PR: `Fixes #<issue number>`.

## Summary

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

## Screenshots

| Before | After |
|--------|-------|
| ... | ... |

## Checklist

- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] I understand that this PR may be closed in case there was no previous discussion or issues. (This doesn't apply to typos!)
- [x] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [x] I've updated the documentation accordingly.
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

nourzakhama2003 committed:
feat(i18n): add Tunisian Arabic (ar-TN) translation

crazywoola commented:
https://github.com/langgenius/dify/blob/58be008676d9513d52835b032561b26f6a5058c1/web/i18n-config/README.md

Some of the configurations are missing.

nourzakhama2003 committed:
fix(i18n): address PR review feedback for ar-TN

nourzakhama2003 committed:
feat(i18n): add Tunisian Arabic (ar-TN) support and translations

- Add server-side language mapping for 'ar-TN' in api/constants/languages.py
- Add/update frontend language config in web/i18n-config/languages.json and language.ts
- Add ar-TN translations in web/i18n/ar-TN/*
- Add temporary debug logging on AccountInterfaceLanguageApi to capture validation issues during testing

Testing and verification steps:
1. Rebuild API: docker compose build api && docker compose up -d api
2. Run frontend in dev: pnpm dev (web folder) and ensure NEXT_PUBLIC_API_PREFIX points to http://localhost/console/api
3. Change language in UI and verify POST /console/api/account/interface-language returns 200 and UI updates

Notes:
- The validation error (400 BAD REQUEST) was caused by the server not recognizing ar-TN, now resolved by adding it to the server languages list.
- Debug logging will be removed or toggled behind NODE_ENV in a follow-up commit if needed

nourzakhama2003 committed:
style(i18n): resolve web style issues for ar-TN by using next/Image and eslint overrides for generated i18n files

QuantumGhost commented on api/constants/languages.py:
```diff
@@ -20,6 +20,7 @@
     "sl-SI": "Europe/Ljubljana",
     "th-TH": "Asia/Bangkok",
     "id-ID": "Asia/Jakarta",
+        "ar-TN": "Africa/Tunis",
```
This line is not properly indented. We use 4 spaces for indentation in Python.

nourzakhama2003 committed:
chore: fix linting and i18n updates

nourzakhama2003 committed:
fix(api): remove redundant try/except in AccountInterfaceLanguageApi (log only, rely on global error handling)

lyzno1 committed:
Merge remote-tracking branch 'origin/main' into i18n-add-ar-tn

lyzno1 committed:
chore: update i18n readme

lyzno1 committed:
chore: clean up eslint configuration by removing commented-out lines

lyzno1 commented:
Please click "resolve" after completing all modifications to resolve all threads.

nourzakhama2003 committed:
chore(i18n): remove unused web/i18n-config/locales.ts per reviewer

nourzakhama2003 committed:
test(i18n): set console_ns.payload in unit tests to simulate request context; fix nesting

nourzakhama2003 committed:
chore: revert unintended changes to CLAUDE.md (restore original file)

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

nourzakhama2003 committed:
Merge branch 'main' into i18n-add-ar-tn

nourzakhama2003 committed:
test(i18n): revert test_account_interface_language.py to prior version (pre i18n console_ns.payload change)

nourzakhama2003 committed:
Merge branch 'main' into i18n-add-ar-tn

nourzakhama2003 committed:
ci: re-run checks

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

nourzakhama2003 committed:
fix(api): fallback to request JSON if console_ns.payload is empty in interface-language endpoint

autofix-ci[bot] committed:
[autofix.ci] apply automated fixes

lyzno1 committed:
test: stabilize account interface language api tests

lyzno1 committed:
remove extra keys

lyzno1 committed:
add missing keys

lyzno1 committed:
Merge remote-tracking branch 'origin/main' into i18n-add-ar-tn

lyzno1 committed:
Update plugin.ts

lyzno1 committed:
Update dataset-pipeline.ts

crazywoola commented on api/controllers/console/workspace/account.py:
Please exclude this file in this PR and the test

nourzakhama2003 replied:
Reverted [account.py](vscode-file://vscode-app/c:/Users/LENOVO/AppData/Local/Programs/Microsoft%20VS%20Code/resources/app/out/vs/code/electron-browser/workbench/workbench.html) to origin/mai

nourzakhama2003 committed:
ci: allow forcing VDB and DB migration tests via PR labels

nourzakhama2003 committed:
ci: explicitly include 'synchronize' for PR triggers to retrigger on pushes to PR branches

nourzakhama2003 committed:
chore: revert changes to account controller and remove account interface language test per review request

nourzakhama2003 committed:
Merge branch 'main' into i18n-add-ar-tn

lyzno1 commented on .github/workflows/main-ci.yml:
Please do not modify this file.

nourzakhama2003 replied:
Reverted [main-ci.yml] to the origin/main version (removed label-based override + explicit synchronize).

lyzno1 reviewed (commented):
The current pull request should only include i18n-related changes. If there are other changes, please submit a separate pull request.

nourzakhama2003 committed:
chore: revert changes to main-ci.yml and signin header per reviewer's request

nourzakhama2003 committed:
Merge branch 'main' into i18n-add-ar-tn

lyzno1 committed:
add translations

Signed-off-by: yyh <yuanyouhuilyz@gmail.com>

lyzno1 reviewed (approved):
Thank you for your contribution!
